### PR TITLE
Refactor masha

### PR DESCRIPTION
**Pull request by maxaalexeeva**
Router + some preprocessing based on what part of the paper the text comes from (prose vs non-prose, e.g., tables of contents, etc).

### CODE REVIEW

File: text_reading/src/main/scala/org/clulab/aske/automates/data/Preprocessor.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    def cleanUp(text: String): String = {
+    //follow up on combining the heading and the body of each section in the paper with "\n" in the DataLoader:
+    //the heading and the body should be connected with a period if the body starts with a capital letter and space otherwise:
+    val cleanerText = text.replaceAll("\n(?=[A-Z])", ". ").replaceAll("\n", " ")
+    val cleanerTextTokenized = cleanerText.split(" ")
+    val numberOfNumbers = cleanerTextTokenized.filter(t => t.forall(_.isDigit)).length
+    val numberProportion = numberOfNumbers.toFloat / cleanerTextTokenized.length
+    val numberThreshold = 0.12
+    cleanerText match {
+      case cleanerText if (numberProportion > numberThreshold && cleanerTextTokenized.length > 10) => {
```
Comment (maxaalexeeva) [inline text_reading/src/main/scala/org/clulab/aske/automates/data/Preprocessor.scala]: changes here, @BeckySharp

File: text_reading/src/main/scala/org/clulab/aske/automates/data/Preprocessor.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        val cleanedUpText = replaceSuspectedPageNumbers(cleanerText)
+        cleanedUpText
+      }
+      case _ => {
+        //todo: add other clean up, e.g., in-paragraph tables
+        replaceEndOfTextPageNumber(cleanerText)}
+    }
+  }
+  //replace on numbers in suspected non-prose sections
+  def replaceSuspectedPageNumbers(text: String): String = text.replaceAll("\\d+\\.?", ". ")
```
Comment (maxaalexeeva) [inline text_reading/src/main/scala/org/clulab/aske/automates/data/Preprocessor.scala]: and the helper methods, @BeckySharp

File: text_reading/src/main/scala/org/clulab/aske/automates/data/Preprocessor.scala (hunk 1 above)
Comment (BeckySharp) [reply text_reading/src/main/scala/org/clulab/aske/automates/data/Preprocessor.scala]: awesome!

File: text_reading/src/main/scala/org/clulab/aske/automates/data/Preprocessor.scala (hunk 2 above)
Comment (BeckySharp) [reply text_reading/src/main/scala/org/clulab/aske/automates/data/Preprocessor.scala]: super awesome!

**Pull request merged**